src/gallium/drivers/tegra/tegra_state.c

```diff
@@ -51,21 +51,49 @@ tegra_set_framebuffer_state(struct pipe_context *pcontext,
 			    const struct pipe_framebuffer_state *framebuffer)
 {
 	struct tegra_context *context = tegra_context(pcontext);
+	struct pipe_framebuffer_state *cso = &context->framebuffer.base;
 	unsigned int i;
+	uint32_t mask = 0;
 
 	for (i = 0; i < framebuffer->nr_cbufs; i++) {
 		struct pipe_surface *ref = framebuffer->cbufs[i];
+		struct tegra_resource *res = tegra_resource(ref->texture);
+		uint32_t rt_params;
+
+		rt_params  = TGR3D_VAL(RT_PARAMS, FORMAT, res->format);
+		rt_params |= TGR3D_VAL(RT_PARAMS, PITCH, res->pitch);
+		rt_params |= TGR3D_BOOL(RT_PARAMS, TILED, res->tiled);
 
-		if (i >= framebuffer->nr_cbufs)
-			ref = NULL;
+		context->framebuffer.rt_params[i] = rt_params;
+		context->framebuffer.bos[i] = res->bo;
+		mask |= 1 << i;
 
-		pipe_surface_reference(&context->framebuffer.base.cbufs[i],
-				       ref);
+		pipe_surface_reference(&cso->cbufs[i], ref);
 	}
 
+	for (; i < cso->nr_cbufs; i++)
+                pipe_surface_reference(&cso->cbufs[i], NULL);
```

Comment:
@digetx: TBH, I copied this logic from [vc4](https://github.com/grate-driver/mesa/blob/master/src/gallium/drivers/vc4/vc4_state.c#L418). My understanding is that `pipe_surface_reference(some_surface, NULL)` removes a reference.